### DISCUSSION TITLE
Confusion surrounding sigmoid/softmax activation in notebook examples

### DISCUSSION THREAD

**Discussion by philnovv** [Project-MONAI/MONAI#6619] [General]
Hi, in the spleen segmentation example here:
https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb

I understand that the output of the model has not been activated by a softmax. Hence, we pass softmax=True to the loss function. But during the validation, when does softmax get applied?

We have:
val_outputs = sliding_window_inference(val_inputs, roi_size, sw_batch_size, model)
val_outputs = [post_pred(i) for i in decollate_batch(val_outputs)]

but post_pred does not apply softmax. What am I missing?

Contrast this to the following segmentation example notebook:
https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_dict.py

where post_pred (here "post_trans") contains the sigmoid activation. Can anyone clarify?

Cheers

**Comment by KumoLiu** [comment]
Hi @philnovv, in the [spleen segmentation tutorial](https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/spleen_segmentation_3d.ipynb) since the output is 2, so if you do softmax->argmax->one-hot which will give the same result by doing argmax->one-hot. You can also add a softmax here. And if you don't add one-hot here, you only need the probability, then you should add softmax here and remove one-hot.
And for [this tutorial](https://github.com/Project-MONAI/tutorials/blob/main/2d_segmentation/torch/unet_training_dict.py), since the output channel is 1, you need to do `AsDiscrete`, so here you must add a sigmoid which can give you a prob between 0 to 1.

Hope this can help you clarify, thanks!
